src/assets/scripts/airport/Route/RouteModel.js

```diff
@@ -0,0 +1,134 @@
+import _uniqueId from 'lodash/uniqueId';
+
+/**
+ * Symbol that divides each route segment
+ *
+ * @property SEGMENT_SEPERATION_SYMBOL
+ * @type {string}
+ * @final
+ */
+const SEGMENT_SEPERATION_SYMBOL = '.';
+
+/**
+ * @class RouteModel
+ */
+export default class RouteModel {
+    /**
+     * 'BETHL.GRNPA1.KLAS'
+     *
+     * @for RouteModel
+     * @constructor
+     * @param routeString {string}
+     */
+    constructor(routeString) {
+        if (typeof routeString === 'undefined' || typeof routeString !== 'string') {
+            return;
+        }
+
+        if (!this._isValidRouteString(routeString)) {
+            // eslint-disable-next-line max-len
+            throw new TypeError(`Invalid routeString passed to RouteModel. Expected a routeString of the shape ORIGIN.BASE.DESTINATION but instead received ${routeString}`);
+        }
+
+        /**
+         * @property
+         * @type {string}
+         */
+        this._id = _uniqueId();
+
+        /**
+         * @property origin
+         * @type {string}
+         * @default ''
+         */
+        this.origin = '';
+
+        /**
+         * @property base
+         * @type {string}
+         * @default ''
+         */
+        this.base = '';
+
+        /**
+         * @property destination
+         * @type {string}
+         * @default ''
+         */
+        this.destination = '';
+
+        return this._init(routeString);
+    }
+
+    /**
+     * A single string that represents the entire route
+     *
+     * @property routeString
+     * @return {string}
+     */
+    get routeString() {
+        return `${this.origin}.${this.base}.${this.destination}`;
+    }
+
+    /**
+     * Lifecycle method. Should be run only once on instantiation
+     *
+     * @for RouteModel
+     * @method _init
+     * @param routeString {string}
+     * @private
+     */
+    _init(routeString) {
+        const { origin, base, destination } = this._extractSegmentNamesFromRouteString(routeString);
+
+        this.origin = origin;
+        this.base = base;
+        this.destination = destination;
+
+        return this;
+    }
+
+    /**
+     * Destroy this instance
+     *
+     * @for RouteModel
+     * @method destroy
+     */
+    destroy() {
+        this._id = '';
+        this.origin = '';
+        this.base = '';
+        this.destination = '';
+    }
+
+    /**
+     *
+     * @for RouteModel
+     * @method _extractSegmentNamesFromRouteString
+     * @param routeString {string}
+     * @return {object}
+     * @private
+     */
+    _extractSegmentNamesFromRouteString(routeString) {
```

Comment:
Two things here I'll start with the second first:  why `_id`?  This is an old habit of mine, you will probably see this in models that will have many instances.  They are not being used at all but provide a way to differentiate instances.  Its helpful when inspecting to have an id for each model.  Who nows, eventually we may want to implement a findById method?  By having the ids in place already, that work would be trivial.

'Why the underscore?'
Remember how I said that Javascript is like the wild west?  This is another example.  the underscore here is a convention for indicating when something is private.  Ill show you some examples of what that means:
```javascript
export default class RouteModel {
    canIPleaseDoThisThing() { 
        return this._youMayProceed();
    }

    _youMayProceed() { }
}

// which would be used as:
const route = new RouteModel();
route.canIPleaseDoThisThing();
```

The above is a common architectural pattern that I use.  Essentially I provide public methods that are really just wrappers for private methods.  The idea of a private method/variable, is that it is used only within that specific class by that specific class.  From the above example, this would be incorrect: `route._youMayProceed();`.  Now there isn't actually anything stopping you from doing that, per-se, but it is not the expected interface.  Same thing goes for private variables.  From the `FixModel` above:
```javascript
get position() {
    return this._fixPosition.position;
}
```
Any external method should not access private methods or variables.  Why is that you ask?  Think about it like this:  I'll give you a way to get into the bank, but you have to speak with the teller (public method) in order for you to get money out of the bank.  How the teller gets that money is of no concern to you (private methods).

Feel free to ask more questions on this one, public/private methods can be confusing.